src/script/lua_api/l_noise.cpp

```diff
@@ -595,3 +599,114 @@ const luaL_reg LuaPcgRandom::methods[] = {
 	luamethod(LuaPcgRandom, rand_normal_dist),
 	{0,0}
 };
+
+///////////////////////////////////////
+/*
+	LuaSecureRandom
+*/
+
+void LuaSecureRandom::fill_buff()
+{
+	if (!porting::secure_rand_fill_buff(m_rand_buff, sizeof(m_rand_buff))) {
+		m_rnd.seed(time(NULL) ^ clock(), 0xda3e39cb94b95bdbULL);
+		size_t i = 0;
+		for (i = 0; i < sizeof(m_rand_buff); i++) {
```

Comment:
I hope nobody ever changes the data type of m_rand_buff... best to use sizeof(m_rand_buff) / sizeof(*m_rand_buff) I think (we don't have a macro to do that?)